Platforms/QemuSbsaPkg/FfaPartitionTest/FfaPartitionTestApp.c

```diff
@@ -209,42 +221,112 @@ FfaPartitionTestAppEntry (
   }
 
   // Then register this test app to receive notifications from the Ffa test SP
-  FfaNotificationBind (FfaTestPartInfo.PartitionId, FFA_NOTIFICATIONS_FLAG_PER_VCPU, 0x1);
+  FfaNotificationBind (FfaTestPartInfo.PartitionId, 0, 0x4);
   if (EFI_ERROR (Status)) {
     DEBUG ((DEBUG_ERROR, "Unable to bind notification with FF-A Ffa test SP (%r).\n", Status));
     goto Done;
   }
 
-  // Next, communicate to the Ffa test SP
-  ZeroMem (&DirectMsgArgs, sizeof (DirectMsgArgs));
-  Status = ArmFfaLibMsgDirectReq (FfaTestPartInfo.PartitionId, 0, &DirectMsgArgs);
+  // Setup the Thermal Service Notification Bits
+  ZeroMem (&DirectMsgArgsEx, sizeof (DirectMsgArgsEx));
+  DirectMsgArgsEx.Arg0 = 0x01; // Setup
+  DirectMsgArgsEx.Arg1 = 0xba7aff2eb1eac765;
+  DirectMsgArgsEx.Arg2 = 0xb610b3a359f64054;
+  DirectMsgArgsEx.Arg3 = 0x03;
+  DirectMsgArgsEx.Arg4 = ((6 << 16) | (0));
+  DirectMsgArgsEx.Arg5 = ((7 << 16) | (1));
+  DirectMsgArgsEx.Arg6 = ((8 << 16) | (2));
+  Status = FfaMessageSendDirectReq2 (FfaTestPartInfo.PartitionId, &FfaNotificationServiceGuid, &DirectMsgArgsEx);
   if (EFI_ERROR (Status)) {
-    DEBUG ((DEBUG_ERROR, "Unable to communicate with FF-A Ffa test SP (%r).\n", Status));
+    DEBUG ((DEBUG_ERROR, "Unable to communicate direct req 2 with FF-A Ffa test SP (%r).\n", Status));
     goto Done;
   }
 
-  DUMP_HEX (DEBUG_INFO, 0, &DirectMsgArgs, sizeof (DirectMsgArgs), "    ");
+  if (DirectMsgArgsEx.Arg0 != 0) {
+    DEBUG ((DEBUG_ERROR, "Command Failed: %x\n", DirectMsgArgsEx.Arg0));
+    goto Done;
+  } else {
+    DEBUG ((DEBUG_ERROR, "Thermal Service Setup Success\n"));

```

Comment:
Was there a particular reason to print these success messages at error level?

```suggestion
    DEBUG ((DEBUG_INFO, "Thermal Service Setup Success\n"));
```

Reply:
No, there was no reason, it's mainly just a copy-paste thing. The entire TestApp is just for verifying the SP functionality and won't ever be used outside of that, as far as I understand. I just wanted print statements to verify things were working as intended. :)

Reply:
That's fine. It is still nice to stick to the expected levels so it matches people's expectations and you'd be surprised where this could end up getting copied/pasted.

Reply:
Agreed. I will update. Looks like I had INFO in some places but not all. That is my fault.